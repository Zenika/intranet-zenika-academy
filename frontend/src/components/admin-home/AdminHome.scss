.admin-home-link{
  margin: 0 1rem 1rem 0
}

.promotionTitle{
  font-weight: bold;
  display: inline;
}

.promosContainer{
  display: flex;
  flex-direction: column;
}

.promoLineContainer{
  margin:.5rem 0;
}

.detailsLink {
  color:blue;
}


