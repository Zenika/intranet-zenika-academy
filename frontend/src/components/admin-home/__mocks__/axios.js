module.exports = {
  get: () =>
    Promise.resolve({
      data: [
        {
          id: 2,
          title: 'PARIS 2',
          city: 'Paris',
          startDate: '2020-10-19T00:00:00.000Z',
          endDate: '2027-01-20T00:00:00.000Z',
          programId: 1,
          createdAt: '2028-11-19T00:00:00.000Z',
          updatedAt: '2028-11-19T00:00:00.000Z',
        },
        {
          id: 3,
          title: 'RENNES',
          city: 'Rennes',
          startDate: '2020-09-14T00:00:00.000Z',
          endDate: '2020-09-20T00:00:00.000Z',
          programId: 1,
          createdAt: '2020-09-24T00:00:00.000Z',
          updatedAt: '2020-09-14T00:00:00.000Z',
        },
        {
          id: 4,
          title: 'GRENOBLE',
          city: 'Grenoble',
          startDate: '2023-06-12T00:00:00.000Z',
          endDate: '2030-07-13T00:00:00.000Z',
          programId: 1,
          createdAt: '2025-05-12T00:00:00.000Z',
          updatedAt: '2012-02-20T00:00:00.000Z',
        },
      ],
    }),
};
