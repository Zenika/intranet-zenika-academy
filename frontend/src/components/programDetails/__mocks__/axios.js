
module.exports = {
  get: () => Promise.resolve({
    data: {
      id: 1,
      title: 'Javascript',
      type: 1,
      content: [
        {
          id: 13,
          title: 'module js',
          type: 2,
          content: [
            {
              id: 14,
              title: 'sous module js',
              type: 3,
              content: [
                {
                  id: 15,
                  title: 'sequence js',
                  type: 4,
                  content: [],
                  createdAt: '2014-02-02T00:00:00.000Z',
                  updatedAt: '2014-02-02T00:00:00.000Z',
                },
              ],
              createdAt: '2014-02-02T00:00:00.000Z',
              updatedAt: '2014-02-02T00:00:00.000Z',
            },
            {
              id: 17,
              title: 'sous module js 2',
              type: 3,
              content: [],
              createdAt: '2014-02-02T00:00:00.000Z',
              updatedAt: '2014-02-02T00:00:00.000Z',
            },
          ],
          createdAt: '2014-02-02T00:00:00.000Z',
          updatedAt: '2014-02-02T00:00:00.000Z',
        },
        {
          id: 16,
          title: 'module 2 js',
          type: 2,
          content: [],
          createdAt: '2014-02-02T00:00:00.000Z',
          updatedAt: '2014-02-02T00:00:00.000Z',
        },
      ],
      createdAt: '2012-12-12T00:00:00.000Z',
      updatedAt: '2012-12-12T00:00:00.000Z',
    },
  }),
};
