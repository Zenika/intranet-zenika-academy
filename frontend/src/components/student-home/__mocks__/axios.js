module.exports = {
  get: () =>
    Promise.resolve({
      data: {
        id: 2,
        title: 'PARIS 2',
        city: 'Paris',
        startDate: '2020-10-19T00:00:00.000Z',
        endDate: '2027-01-20T00:00:00.000Z',
        programId: 1,
        createdAt: '2028-11-19T00:00:00.000Z',
        updatedAt: '2028-11-19T00:00:00.000Z',
      },
    }),
};
