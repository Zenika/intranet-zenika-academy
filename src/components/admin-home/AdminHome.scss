.admin-home-link{
  margin: 0 1rem 1rem 0
}

.detailsLink {
  color:blue;
}

.promoLineContainer{
  margin:.5rem 0;
}

.promosContainer{
  display: flex;
  flex-direction: column;
}
