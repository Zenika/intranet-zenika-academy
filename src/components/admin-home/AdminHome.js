import React, { Component } from 'react';
import axios from 'axios';
import Moment from 'react-moment';
import './AdminHome.scss';

class AdminHome extends Component {
  _isMounted = false;

  constructor(props) {
    super(props);
    this.state = { promotions: {} };
  }

  componentDidMount() {
    this._isMounted = true;
    axios.get('http://localhost:4000/api/promotions')
      .then((res) => {
        const promotions = res.data;
        if (this._isMounted) {
          this.setState({ promotions });
        }
      });
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  render() {
    // eslint-disable-next-line react/destructuring-assignment
    const promotions = this.state.promotions.length > 0 ? this.state.promotions : [];
    return (
      <>
        <h1 className="title is-2 mbmd">Bienvenue</h1>

        <a href="/admin/promo/create" className="button is-primary admin-home-link">Créer une promotion</a>
        <a href="/admin/program/create" className="button is-primary admin-home-link">Créer un programme</a>
        <a href="/admin/user/create" className="button is-primary admin-home-link">Créer un utilisateur</a>

        <h2 className="title is-4 mbmd">Liste des promotions :</h2>
        <ul className="promosContainer">
          {promotions.map((promotion) => (
            <li className="promoLineContainer" key={promotion.id}>
              <h1 className="promotionTitle">{promotion.title}</h1>
              {' du '}
              <Moment format="DD/MM/YYYY">{promotion.startDate}</Moment>
              {' au '}
              <Moment format="DD/MM/YYYY">{promotion.endDate}</Moment>
              {' à '}
              {promotion.city}
              { ' ' }
              <a href={`/admin/promo/${promotion.id}/details`} className="detailsLink">détails...</a>
            </li>
          ))}
        </ul>
      </>
    );
  }
}

export default AdminHome;
