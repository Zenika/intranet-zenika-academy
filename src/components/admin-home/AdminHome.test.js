import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import axios from 'axios';
import Adapter from 'enzyme-adapter-react-16';
import AdminHome from './AdminHome';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('axios');

describe('AdminHome component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AdminHome />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches snapshot', () => {
    const component = renderer.create(<AdminHome />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('contains a title "Bienvenue"', () => {
    const app = shallow(<AdminHome />);
    expect(app.containsMatchingElement(<h1>Bienvenue</h1>)).toEqual(true);
  });
  describe('when rendered', () => {
    it('should fetch promotions', () => {
      const getSpy = jest.spyOn(axios, 'get');
      shallow(
        <AdminHome />,
      );
      expect(getSpy).toBeCalled();
    });
  });
});
