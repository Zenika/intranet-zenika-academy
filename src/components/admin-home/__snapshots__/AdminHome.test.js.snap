// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHome component matches snapshot 1`] = `
Array [
  <h1
    className="title is-2 mbmd"
  >
    Bienvenue
  </h1>,
  <a
    className="button is-primary"
    href="/admin/promo/create"
  >
    Créer une promotion
  </a>,
  <a
    className="button is-primary"
    href="/admin/program/create"
  >
    Créer un programme
  </a>,
  <a
    className="button is-primary"
    href="/admin/user/create"
  >
    Créer un utilisateur
  </a>,
  <h2
    className="title is-4 mbmd"
  >
    Liste des promotions :
  </h2>,
  <ul
    className="promosContainer"
  />,
]
`;
